### ISSUE TITLE
Brave news doesn't refresh after unfollowing `Top sources `

### ISSUE THREAD

**Issue by MadhaviSeelam** [brave/brave-browser#35052]
## Description 

Found while testing https://github.com/brave/brave-browser/issues/34593.  After unfollowing `Top sources`, had to manually refresh Brave news feed.  It should show `New content available. Reload?`  message. 

## Steps to Reproduce

   1. Install `1.62.122`
   2. launch Brave
   3. enable Brave news
   4. follow few channels (Education & Film & TV) and Publishers (CNN & The New York Times)
   5. close Brave news dialog
   6. clicked `News` button to scroll down into the feed
   7. clicked `New content available. Reload?` in the feed
   8. unfollow `Top Sources` via `customize` button
   9. close Brave news dialog to return to feed
   
   
## Actual result:

https://github.com/brave/brave-browser/assets/98358127/453a599c-6310-486f-a5bc-097f467d2645

## Expected result:
`New content available. Reload?`  message should show on the feed to refresh the page.

## Reproduces how often: 

Easily

## Brave version (brave://version info)

```
Brave | 1.62.122 Chromium: 120.0.6099.144 (Official Build) beta (64-bit)
-- | --
Revision | cfddebe77d394064c472fda64afcd9fbed34ceb4
OS | Windows 11 Version 22H2 (Build 22621.2861)
```

## Version/Channel Information:

- Can you reproduce this issue with the current release? N/A
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? No

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

@fallaciousreasoning @LorenzoMinto 
cc: @stephendonner @rebron

**Comment by fallaciousreasoning** [comment]
This might have been fixed by https://github.com/brave/brave-core/pull/21439

**Comment by stephendonner** [comment]
> This might have been fixed by [brave/brave-core#21439](https://github.com/brave/brave-core/pull/21439)

Indeed; by `Case 3`, there.

**Comment by MadhaviSeelam** [comment]
Working as expected
<img width="942" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/4742a3ad-3407-4d5d-b720-863da395fa57">
